### ISSUE TITLE
Component rules not applied for components extending a common base component

### ISSUE THREAD

**Issue by bdmac** [netguru/eslint-plugin-ember#27]
The various component level rules do not get applied if you extend another application level component.

For example this shows no linter errors when it should:
```javascript
export default MyComponent.extend({
  foo: 'bar',
   store: inject.service()
});
```

Whereas this does:
```javascript
export default Component.extend({
  foo: 'bar',
   store: inject.service()
});
```

**Comment by michalsnik** [comment]
Hi @bdmac, yes it is a problem.

Currently we distinct routes from controllers and components by checking name of the module. If you create one by extending other component we simply can not distinguish what it is actually.

I'll leave this issue marked as `in consideration` and I'll get back to you as soon as I find the solution to this problem. I have an idea on my mind but I need to check this.

If you have any ideas - feel free to help with this one :)

**Comment by bdmac** [comment]
I have ideas but I'm not sure how eslint works really. One would be to check the directory of the file under analysis and see if it has `component` in the hierarchy. A bit hacky but given ember's directory structure should work. Another option is to walk included extended modules but again not sure if eslint has access to that.

**Comment by michalsnik** [comment]
I did a quick research, and it's totally possible based on the file path, and it would be 100% testable, so I think it's the proper way of handling this scenario.

What are your thoughts on this one guys? @jbandura @netes @bardzusny 

Getting file path in inside rule is really straightforward:
```
context.getFilename();
```

and setting filename in test looks like this:
```
{
  filename: 'example-app/components/path/to/some-component.js',
  code: `export default CustomComponent.extend({
    // ...
  });`,
  parserOptions: { ecmaVersion: 6, sourceType: 'module' },
},
```

**Comment by michalsnik** [comment]
@bdmac Released in 3.1.0 🚀

**Comment by bdmac** [comment]
🎆
